skimage/util/dask_compatibility.py

```diff
@@ -0,0 +1,493 @@
+        will not be broken down further.
+
+    Returns
+    -------
+    smaller_chunk : iterable of ints
+        Same length as ``shape``, this corresponds to an appropriate value for
+        ``dask.array.from_array(..., chunks=smaller_chunk)``.
+    """
+    ram = psutil.virtual_memory().total * mem_frac
+
+    # Size of each array element
+    bytes_per_entry = np.dtype(dtype).itemsize
+
+    # Start with current shape, decrease iteratively as necessary
+    smaller_chunk = shape
+    preserve_dims_under *= 2
+
+    while (bytes_per_entry * _total_elements(smaller_chunk)) > ram:
+        smaller_chunk = tuple(dim // 2
+                              if dim > preserve_dims_under
+                              else dim
+                              for dim in smaller_chunk)
+
+    return smaller_chunk
+
+
+def asanyarray(arr_like):
+    """
+    Convert array-like input to array, passing dask arrays through.
+
+    Parameters
+    ----------
+    arr_like : array-like
+        Array-like input object to be converted to an array. Can be a NumPy
+        array, a ``dask.array``, or any other array-like object.
+
+    Returns
+    -------
+    arr : array
+        If ``arr_like`` was a ``dask.array`` object it is passed unmodified,
+        otherwise a NumPy array is returned.
+
+    Notes
+    -----
+    This function is intended to replace ``np.asanyarray`` in the package so
+    Dask arrays can be passed without conversion back into NumPy objects.
+    """
+    if isinstance(arr_like, da.core.Array):
+        return arr_like
+    else:
+        return np.asanyarray(arr_like)
+
+
+def asarray(arr_like):
+    """
+    Convert array-like input to array, passing dask arrays through.
+
+    Parameters
+    ----------
+    arr_like : array-like
+        Array-like input object to be converted to an array. Can be a NumPy
+        array, a ``dask.array``, or any other array-like object.
+
+    Returns
+    -------
+    arr : array
+        If ``arr_like`` was a ``dask.array`` object it is passed unmodified,
+        otherwise a NumPy array is returned.
+
+    Notes
+    -----
+    This function is intended to replace ``np.asarray`` in the package so
+    Dask arrays can be passed without conversion back into NumPy objects.
+    """
+    if isinstance(arr_like, da.core.Array):
+        return arr_like
+    else:
+        return np.asarray(arr_like)
+
+
+def dtype_conversion_inplace(arr, new_dtype):
+    """
+    Convert dtype of NumPy or dask array inplace.
+
+    Parameters
+    ----------
+    arr : dask or NumPy array
+        Input array, NumPy or Dask arrays supported.
+    new_dtype : NumPy dtype object
+        Valid destination dtype for ``arr``.
+
+    Returns
+    -------
+    arr : dask or NumPy array
+        Output array, converted to dtype ``new_dtype``.
+
+    Notes
+    -----
+    This function is intended to replace ``np.array(..., copy=False)`` as
+    Dask has no equivalent functionality.
```

Comment:
Should there be such functionality?  We can add things to dask.array if they're useful here.